### PR TITLE
Reflection / IT-Academy / JC1-117-22-dev / Haretski Aliaksei / Done all tasks 

### PR DESCRIPTION
**Pull request by haretski-aliaksei**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [ ] fixed
- [ ] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/reflection/reflection_utils/ReflectionUtils.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+public class ReflectionUtils {
+    /**
+     * The method takes "Class", the name of the field and returns "Field".
+     *
+     * @param className incoming parameter
+     * @param fieldName incoming parameter
+     * @return "Field"
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/reflection/reflection_utils/ReflectionUtils.java:13]: Use `@see`

File: src/main/java/com/rakovets/course/java/core/practice/reflection/reflection_utils/ReflectionUtils.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+     */
+    public Field getFieldFromName(Class className, String fieldName) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/reflection/reflection_utils/ReflectionUtils.java:15]:
```suggestion
    public Field getFieldByName(Class clazz, String fieldName) {
```

File: src/main/java/com/rakovets/course/java/core/practice/reflection/reflection_utils/ReflectionUtils.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        try {
+            Field field = className.getDeclaredField(fieldName);
+            return field;
+        } catch (NoSuchFieldException e) {
+            throw new RuntimeException(e);
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/reflection/reflection_utils/ReflectionUtils.java:20]: use Reflection API exceptions OR throw custom exception

**Review by rakovets — CHANGES_REQUESTED** [review]
Where is tests OR demo?

File: src/main/java/com/rakovets/course/java/core/practice/reflection/reflection_utils/ReflectionUtils.java (hunk 2 above)
Comment (haretski-aliaksei) [reply src/main/java/com/rakovets/course/java/core/practice/reflection/reflection_utils/ReflectionUtils.java:15]: Fixed

**Comment by haretski-aliaksei** [comment]
Fixed PR: https://github.com/rakovets/course-java-core/pull/1117

**Pull request closed without merging**